woocommerce/woocommerce-ios
# [Mobile Payments] Add CardReaderSettingsKnown(Readers)ViewModel

This is the next PR in a series of PRs for the "automatically reconnect to known card readers" feature

This PR adds a new view model CardReaderSettingsKnownViewModel that will return shouldShow true when we are searching for readers AND we have one or more remembered readers in app settings.

Although very similar to CardReaderSettingsUnknownViewModel, a separate view model is desired to keep that view model (and view) single-concern, i.e. because
- the view controller this view model pairs with needs to initiate search automatically
- this view model needs to connect automatically if discovery yields a known reader ID
- the view controller this view model will be paired with may soon need different content (than the CardReaderSettingsUnknownViewController has today) since the merchant will have connected successfully to a reader already

The next PR will add that new view controller.

Note that we have not added this view model to CardReaderSettingsViewModelsOrderedList. That comes in a later PR.

Note that removing the older known reader logic from places like the CardReaderSettingsUnknownViewModel comes in a later PR in this sequence. The sequence of planned PRs can be found in the last comment on #3559 

To test:

No user facing changes for this viewmodel
Ensure linting passes
Ensure unit tests pass

Update release notes:
